### ISSUE TITLE
How to design loss terms of various generation models — RuifMa

### ISSUE THREAD

**Issue by RuifMaxx** [RuifMaxx/RuifMaxx.github.io#14]
https://ruifmaxx.github.io/2020/12/14/LOSS/ 

 VAE ELBO```pythondef log_normal_pdf(x, mean, logvar): const = torch.from_numpy(np.array([2. * np.pi])).float().to(x.device) const = torch.log(const) return -.5 * (const + logvar + (x - mean) ** 2. / torch.exp(logvar))
